src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,145 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private static final int GROW_FACTOR = 2;
+
+    private Node<K, V>[] table;
+    private double loadFactor;
+    private int threshold;
+    private int size;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY, DEFAULT_LOAD_FACTOR);
+    }
+
+    public MyHashMap(int initialCapacity, double loadFactor) {
+        this.table = new Node[initialCapacity];
+        this.loadFactor = loadFactor;
+        this.threshold = (int) (initialCapacity * loadFactor);
+    }
+
+    @Override
+    public int getSize() {
+        return size;
+    }
 
     @Override
     public void put(K key, V value) {
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
 
+        int hash = hash(key);
+        int index = Math.floorMod(hash, table.length);
+        if (table[index] == null) {
+            table[index] = new Node<>(hash, key, value, null);
```

Comment:
will be better if add newNode(K key, V value) method which will return new node . that method will help unload all methods for "creation" and can be use multiple time
```suggestion
       newNode(K key, V value)
```

Reply:
Or make another constructor in Node class